server/src/main/java/org/elasticsearch/action/bulk/BulkItemResponse.java

```diff
@@ -505,7 +506,9 @@ public void writeThin(StreamOutput out) throws IOException {
     }
 
     private void writeResponseType(StreamOutput out) throws IOException {
-        if (response instanceof IndexResponse) {
+        if (response instanceof SimulateIndexResponse) {
+            out.writeByte((byte) 4);
```

Comment:
Will this blow up older ES versions that don't support this type of response?

Reply:
I don't think it will -- they couldn't wind up with a SimulateIndexResponse if they don't know how to create a SimulateIndexRequest, right?